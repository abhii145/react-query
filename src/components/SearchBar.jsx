import { useState } from "react"
import { AiOutlineClose } from "react-icons/ai"

const SearchBar = ({ searchQuery, onSearchChange }) => {
  const [inputValue, setInputValue] = useState(searchQuery)

  const handleInputChange = (e) => {
    const value = e.target.value
    setInputValue(value)
    onSearchChange(value)
  }

  const clearInput = () => {
    setInputValue("")
    onSearchChange("")
  }

  return (
    <div className="mb-6 relative">
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        placeholder="Search products..."
        className="w-full p-3 border border-gray-300 rounded-md text-sm focus:ring-2 focus:ring-blue-500 focus:outline-none"
      />
      {inputValue && (
        <AiOutlineClose
          onClick={clearInput}
          className="absolute right-3 top-1/2 transform -translate-y-1/2 cursor-pointer text-gray-500"
        />
      )}
    </div>
  )
}

export default SearchBar
