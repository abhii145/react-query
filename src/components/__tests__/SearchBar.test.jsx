import { render, screen, fireEvent } from "@testing-library/react"
import SearchBar from "../SearchBar"
import { it, expect, describe, vi } from "vitest"

describe("SearchBar", () => {
  const onSearchChange = vi.fn()

  it("renders input with initial value", () => {
    render(<SearchBar searchQuery="test" onSearchChange={onSearchChange} />)
    const input = screen.getByPlaceholderText("Search products...")
    expect(input.value).toBe("test")
  })

  it("calls onSearchChange on input change", () => {
    render(<SearchBar searchQuery="" onSearchChange={onSearchChange} />)
    const input = screen.getByPlaceholderText("Search products...")
    fireEvent.change(input, { target: { value: "new value" } })
    expect(onSearchChange).toHaveBeenCalledWith("new value")
  })

  it("clears input when clear button is clicked", () => {
    render(<SearchBar searchQuery="test" onSearchChange={onSearchChange} />)
    const input = screen.getByPlaceholderText("Search products...")
    const clearButton = screen.getByTestId("clear-button")
    fireEvent.click(clearButton)
    expect(input.value).toBe("")
    expect(onSearchChange).toHaveBeenCalledWith("")
  })
})
