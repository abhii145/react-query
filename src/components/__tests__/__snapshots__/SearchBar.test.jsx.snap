// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-6 relative"
  >
    <input
      class="w-full p-3 border border-gray-300 rounded-md text-sm focus:ring-2 focus:ring-blue-500 focus:outline-none"
      placeholder="Search products..."
      type="text"
      value="test"
    />
    <svg
      class="absolute right-3 top-1/2 transform -translate-y-1/2 cursor-pointer text-gray-500"
      data-testid="clear-button"
      fill="currentColor"
      fill-rule="evenodd"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M799.855 166.312c.023.007.043.018.084.059l57.69 57.69c.041.041.052.06.059.084a.118.118 0 0 1 0 .069c-.007.023-.018.042-.059.083L569.926 512l287.703 287.703c.041.04.052.06.059.083a.118.118 0 0 1 0 .07c-.007.022-.018.042-.059.083l-57.69 57.69c-.041.041-.06.052-.084.059a.118.118 0 0 1-.069 0c-.023-.007-.042-.018-.083-.059L512 569.926 224.297 857.629c-.04.041-.06.052-.083.059a.118.118 0 0 1-.07 0c-.022-.007-.042-.018-.083-.059l-57.69-57.69c-.041-.041-.052-.06-.059-.084a.118.118 0 0 1 0-.069c.007-.023.018-.042.059-.083L454.073 512 166.371 224.297c-.041-.04-.052-.06-.059-.083a.118.118 0 0 1 0-.07c.007-.022.018-.042.059-.083l57.69-57.69c.041-.041.06-.052.084-.059a.118.118 0 0 1 .069 0c.023.007.042.018.083.059L512 454.073l287.703-287.702c.04-.041.06-.052.083-.059a.118.118 0 0 1 .07 0Z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
